### ISSUE TITLE
make json_benchmarks no longer working in 2.0.4

### ISSUE THREAD

**Issue by whackashoe** [nlohmann/json#310]
```
$ make json_benchmarks 
cd benchmarks/files/numbers ; python generate.py
g++ -std=c++11  -DNDEBUG -O3 -flto -I src -I benchmarks benchmarks/benchmarks.cpp  -o json_benchmarks
./json_benchmarks
parse jeopardy.json                           1  2970652448 ns/op
parse canada.json                            50    44576518 ns/op
parse citm_catalog.json                      50    22833206 ns/op
parse twitter.json                          100    12558957 ns/op
terminate called after throwing an instance of 'std::invalid_argument'
  what():  parse error - unexpected end of input; expected ']'
make: *** [json_benchmarks] Aborted
make: *** Deleting file `json_benchmarks'
```

2daab5a4c24daa447a9d636b5c62121933fb84f1 is culprit.

**Comment by nlohmann** [comment]
After reading a line from a stream, the newline symbol was added at the wrong place. The issue is fixed. The fix shall be released as 2.0.5.

**Comment by nlohmann** [comment]
Thanks for reporting!
